hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/steps/NodeStakeUpdates.java

```diff
@@ -173,14 +175,17 @@ public static boolean isNextStakingPeriod(
     }
 
     private void keyNewRoster(
-            @NonNull final HandleContext handleContext, @NonNull final WritableRosterStore rosterStore) {
+            @NonNull final HandleContext handleContext,
+            @NonNull final WritableRosterStore rosterStore,
+            @NonNull final TssMetrics tssMetrics) {
         final var nodeStore = handleContext.storeFactory().readableStore(ReadableNodeStore.class);
         final var newCandidateRoster = nodeStore.newRosterFromNodes();
 
         if (!Objects.equals(newCandidateRoster, rosterStore.getCandidateRoster())
                 && !Objects.equals(newCandidateRoster, rosterStore.getActiveRoster())) {
+            tssMetrics.updateCandidateRosterLifecycle(Instant.now().getEpochSecond());
```

Comment:
We should use `InstantSource` for the timestamp here, since we have EmbeddedHedera in the test framework now